docs/spec/v1beta2/helmrepositories.md

```diff
@@ -553,3 +553,170 @@ reconcile](#triggering-a-reconcile).
 
 [typical-status-properties]: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
 [kstatus-spec]: https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus
+
+## OCI Hem Repositories
```

Comment:
I think we need to link this from the top level section and redo all headers. After the top level title, we could link to:
- HTTP/S Helm Repositories
- OCI Helm Repositories

@hiddeco may have a different idea.

Reply:
As I went through great lengths to not duplicate any information in this document, I feel that this could indeed be better structured. My proposal would be the following:

- Delete this whole new section.
- Change introduction of document ("[..] API defines a Source to produce an Artifact [..]") to highlight there are two types.
- Add sub-sections to "examples":

	1. Example Helm HTTP/S repository
	2. Example Helm OCI repository

  Where the OCI section contains the same outline of information around the reconciler as what currently exists for HTTP/S.
- Add section for "Type" to "Writing a HelmRepository spec"
- Add note to "TLS authentication" that it only applies to non-OCI.
- Add a note about image pull secrets for OCI being worked on in https://github.com/fluxcd/flux2/pull/2601
- Instead of duplicating the Conditions section, find a way to highlight that the OCI type just has a `Ready` condition and no artifact.

Reply:
Thanks for this! I have updated the doc. I did not add the part regarding the image pull secret though, I felt like it is better to update the doc when we implement it.